### PR TITLE
Fix for export failing when product_id missing from cataloginventory_stock_item

### PR DESCRIPTION
**Pull request by discountscott**
### Description (*)

Export from admin panel would fail if a `product_id` listed in `catalog_product_entity` is not in the `cataloginventory_stock_item` table. This shouldn't be the case, but one of our sites encountered this issue.

This PR prevents the export from failing with a 500 error.

### Fixed Issues (if relevant)

1. Fixes OpenMage/magento-lts#<issue_number>

### Manual testing scenarios (*)

1. `catalog_product_entity` row listed without a SKU (for unknown reason)
2. `entity_id` not referenced in `cataloginventory_stock_item` `product_id` column
3. Generates 500 error on Import/Export -> Export -> products csv

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

**Comment by kiatng** [comment]
Instead of silently continuing processing the export, is it better to `Mage::throwException()` to alert the user of the issue?

**Comment by discountscott** [comment]
There's no meaningful way for an end-user to correct the problem in the admin panel. As the product_id that does not exist in inventory_stock cannot be seen from the "manage products" screen.

Maybe best to still continue but log the issue?

File: app/code/core/Mage/ImportExport/Model/Export/Entity/Product.php [hunk 1]
```diff
@@ -926,7 +926,9 @@ protected function _prepareExport()
                         $dataRow[self::COL_TYPE]     = null;
                     } else {
                         $dataRow[self::COL_STORE] = null;
-                        $dataRow += $stockItemRows[$productId];
+                        if ($stockItemRows[$productId]) {
```
Comment (kiatng) [inline app/code/core/Mage/ImportExport/Model/Export/Entity/Product.php:929]: Agree, export should continue, garbage in garbage out. Is this better:

```suggestion
                        if (!empty($stockItemRows[$productId])) {
```

**Comment by kiatng** [comment]
What's the underlying error that threw 500?

**Comment by discountscott** [comment]
> What's the underlying error that threw 500?

 Unsupported operand type on this line ```$dataRow += $stockItemRows[$productid];```

File: app/code/core/Mage/ImportExport/Model/Export/Entity/Product.php [hunk 2]
```diff
@@ -926,7 +926,7 @@ protected function _prepareExport()
                         $dataRow[self::COL_TYPE]     = null;
                     } else {
                         $dataRow[self::COL_STORE] = null;
-                        if (!empty($stockItemRows[$productId])) {
+                        if ((!empty($stockItemRows[$productId]))&&(is_numeric($stockItemRows[$productId]))) {
```
Comment (fballiano) [inline app/code/core/Mage/ImportExport/Model/Export/Entity/Product.php:929]:
```suggestion
                        if (!empty($stockItemRows[$productId]) && is_numeric($stockItemRows[$productId])) {
```

File: app/code/core/Mage/ImportExport/Model/Export/Entity/Product.php [hunk 3]
```diff
@@ -926,7 +926,9 @@ protected function _prepareExport()
                         $dataRow[self::COL_TYPE]     = null;
                     } else {
                         $dataRow[self::COL_STORE] = null;
-                        $dataRow += $stockItemRows[$productId];
+                        if (!empty($stockItemRows[$productId])) {
+                            $dataRow += $stockItemRows[$productId];
```
[the remark this answers is not in the document]
Comment (discountscott) [reply app/code/core/Mage/ImportExport/Model/Export/Entity/Product.php:930]: my mistake, it's supposed to be an array...

[the remark this answers is not in the document]
Comment (fballiano) [reply app/code/core/Mage/ImportExport/Model/Export/Entity/Product.php:930]: phpstan still not happy :-(

[the remark this answers is not in the document]
Comment (discountscott) [reply app/code/core/Mage/ImportExport/Model/Export/Entity/Product.php:930]: I just fixed it `$stockItemRows[$productid]` is an array. changed to is_array

**Review by kiatng — APPROVED** [review]
LGTM

**Review by fballiano — APPROVED** [review]

**Pull request merged**